pkgs/development/tools/godot/4/mono/default.nix

```
@@ -0,0 +1,208 @@
+, withPulseaudio ? true
+, withDbus ? true
+, withSpeechd ? true
+, withFontconfig ? true
+, withUdev ? true
+, withTouch ? true
+, dotnet-sdk
+, mono
+, dotnet-runtime
+, callPackage
+}:
+
+assert lib.asserts.assertOneOf "withPrecision" withPrecision [ "single" "double" ];
+
+let
+  mkSconsFlagsFromAttrSet = lib.mapAttrsToList (k: v:
+    if builtins.isString v
+    then "${k}=${v}"
+    else "${k}=${builtins.toJSON v}");
+in
+stdenv.mkDerivation rec {
+  pname = "godot4-mono";
+  version = "4.2.1-stable";
+  commitHash = "b09f793f564a6c95dc76acc654b390e68441bd01";
+
+  nugetDeps = mkNugetDeps { name = "deps"; nugetDeps = import ./deps.nix; };
+
+  shouldConfigureNuget = true;
+
+  nugetSource =
+    mkNugetSource {
+      name = "${pname}-nuget-source";
+      description = "A Nuget source with dependencies for ${pname}";
+      deps = [ nugetDeps ];
+  };
+
+  nugetConfig = writeText "NuGet.Config" ''
+    <?xml version="1.0" encoding="utf-8"?>
+    <configuration>
+      <packageSources>
+        <add key="${pname}-deps" value="${nugetSource}/lib" />
+      </packageSources>
+    </configuration>
+  '';
+
+  src = fetchFromGitHub {
+    owner = "godotengine";
+    repo = "godot";
+    rev = commitHash;
+    hash = "sha256-Q6Og1H4H2ygOryMPyjm6kzUB6Su6T9mJIp0alNAxvjQ";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    autoPatchelfHook
+    installShellFiles
+    python3
+    mono
+    dotnet-sdk
+    dotnet-runtime
+  ];
+
+  buildInputs = [
+    scons
+  ];
+
+  runtimeDependencies = [
+    vulkan-loader
+    libGL
+    libX11
+    libXcursor
+    libXinerama
+    libXext
+    libXrandr
+    libXrender
+    libXi
+    libXfixes
+    libxkbcommon
+    alsa-lib
+    mono
+    dotnet-sdk
+    dotnet-runtime
+  ]
+  ++ lib.optional withPulseaudio libpulseaudio
+  ++ lib.optional withDbus dbus
+  ++ lib.optional withDbus dbus.lib
+  ++ lib.optional withSpeechd speechd
+  ++ lib.optional withFontconfig fontconfig
+  ++ lib.optional withFontconfig fontconfig.lib
+  ++ lib.optional withUdev udev;
+
+  enableParallelBuilding = true;
+
+  # Set the build name which is part of the version. In official downloads, this
+  # is set to 'official'. When not specified explicitly, it is set to
+  # 'custom_build'. Other platforms packaging Godot (Gentoo, Arch, Flatpack
+  # etc.) usually set this to their name as well.
+  #
+  # See also 'methods.py' in the Godot repo and 'build' in
+  # https://docs.godotengine.org/en/stable/classes/class_engine.html#class-engine-method-get-version-info
+  BUILD_NAME = "nixpkgs";
+
+  # Required for the commit hash to be included in the version number.
+  #
+  # `methods.py` reads the commit hash from `.git/HEAD` and manually follows
+  # refs. Since we just write the hash directly, there is no need to emulate any
+  # other parts of the .git directory.
+  #
+  # See also 'hash' in
+  # https://docs.godotengine.org/en/stable/classes/class_engine.html#class-engine-method-get-version-info
+  preConfigure = ''
```

```suggestion
  preConfigure = ''
```
Order attributes by derivation phases order. Which is the execution order of hooks.  For looking up order: https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

(Errors will also show in sequence.)